### PR TITLE
tmux-fingers: 1.0.1 -> 2.1.1

### LINKED ISSUE #279999: Package request: tmux-fingers
**Project description**
copy pasting in terminal with vimium/vimperator like hints.

**Metadata**

* homepage URL: https://github.com/Morantron/tmux-fingers
* source URL: https://github.com/Morantron/tmux-fingers
* license: mit
* platforms: unix

**Additional information**
Tmux-thumb is available, but it has certain acknowledged issues, as indicated in https://github.com/fcsonline/tmux-thumbs/issues/76. 
In comparison, the original tmux-finger proves to be more reliable for my needs.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by WilliamHsieh**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Resolve https://github.com/NixOS/nixpkgs/issues/279999

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/misc/tmux-plugins/default.nix [hunk 1]
```diff
@@ -244,23 +244,25 @@ in rec {
...
-      fetchSubmodules = true;
+      rev = "${version}";
+      sha256 = "sha256-1YMh6m8M6FKf8RPXsOfWCVC5CXSr/MynguwkG7O+oEY=";
     };
-    nativeBuildInputs = [ pkgs.makeWrapper ];
+    nativeBuildInputs = [ pkgs.makeWrapper pkgs.crystal pkgs.shards ];
     postInstall = ''
-      for f in config.sh tmux-fingers.sh setup-fingers-mode-bindings.sh; do
-      wrapProgram $target/scripts/$f \
+      shards build --production
```
Comment (Mic92) [inline pkgs/misc/tmux-plugins/default.nix:257]: Is it intended that the package also contains tests and source code in addition to the binary?

Comment (WilliamHsieh) [reply pkgs/misc/tmux-plugins/default.nix:257]: Oh, actually only the binary is needed for this plugin.
Is there a cleaner way to achieve this other then adding `rm -rf $target/*` after the build?

Comment (Mic92) [reply pkgs/misc/tmux-plugins/default.nix:257]: Looks like the builder doesn't allow anything else. So this would be the most pragmatic way.

Comment (WilliamHsieh) [reply pkgs/misc/tmux-plugins/default.nix:257]: done

File: pkgs/misc/tmux-plugins/default.nix [hunk 2]
```diff
@@ -244,23 +244,26 @@ in rec {
...
+      rm -rf $target/*
```
Comment (Mic92) [inline pkgs/misc/tmux-plugins/default.nix:258]: We also need to drop these dotfiles, right?

```suggestion
      rm -rf $target/* $target/.*
```

Comment (WilliamHsieh) [reply pkgs/misc/tmux-plugins/default.nix]: ah sure, thanks

**Pull request merged**